docs/_posts/2022-05-06-writing-cli-tools-with-typescript.md

```diff
@@ -0,0 +1,187 @@
+
+Preferably, we also want to have a single, packaged binary as the result of building our project, rather than a script file, to reduce installation overhead on part of the user. 
+
+To assure high code quality we also typically want to include code formatting, linting, and tests.
+
+Preferably, we also want to have a single, packaged binary as the result of building our project rather than a script file to reduce installation overhead on part of the user. 
+
+To assure high code quality we also typically want to include code formatting, linting, and tests.
+
+## Basic development flow
+
+To hit the ground running, I created a starter project which already contains all required boilerplate and dependencies for running, formatting, linting, testing the code and packaging it to a binary. 
+
+> ℹ️ The starter project can be found on my [GitHub](https://github.com/fkurz/typescript-cliutil-starter) and via [npm](https://www.npmjs.com/package/@fkurz/typescript-cliutil-starter/). 
+
+```bash
+git clone git@github.com:fkurz/typescript-cliutil-starter.git my-cli
+cd my-cli 
+npm i
+```
+
+Our starter project uses 
+
+- the [*commander*](https://www.npmjs.com/package/commander) package for argument parsing and executing commands, 
+- code linting and formatting with [eslint](https://eslint.org/) and [prettier](https://prettier.io/), 
+- [Jest](https://jestjs.io/) as test runner and assertion library, and 
+- [pkg](https://github.com/vercel/pkg) to build a stand-alone binary.
+
+To illustrate the development flow, we now add an example subcommand and test it. 
+
+For example, we could add a command *yell* which is exactly like the *say* command (that already exists in the project) but a bit less subtle.
+
+To do this, we add a file *src/yell.ts* and then add the following code.
+
+```typescript
+// src/yell.ts
+import { Command } from "commander";
+
+export default (): Command =>
+ new Command()
+  .name("yell")
+  .description("Say the word passed as the first argument but louder")
+  .argument("<word>")
+  .action((word: string) => console.log(word.toLocaleUpperCase()));
+```
+
+
+
+We also have to register our yell command with the main CLI command in src/cmd.ts, i.e. add an import statement for our subcommand and register it using *Commander.addCommand* 
+
+```typescript
+// src/cmd.ts
+// … 
+import buildYellCmd from "./yell";
+// …
+export default (): Command => {
+ const command = new Command()
+  .option("-g, --greet", `Say ${HELLO}`, false)
+  .addCommand(buildSayCmd())
+  .addCommand(buildYellCmd())
+  .addHelpCommand()
+  .showHelpAfterError();
+```
```

Comment:
Either there's something missing (return statement, closing brace) or you wanted to write a lambda expression.

```suggestion
export default (): Command =>
 new Command()
  .option("-g, --greet", `Say ${HELLO}`, false)
  .addCommand(buildSayCmd())
  .addCommand(buildYellCmd())
  .addHelpCommand()
  .showHelpAfterError();
```
```